source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -284,29 +289,57 @@ FaultFilter::delayDuration(const Http::RequestHeaderMap& request_headers) {
   return ret;
 }
 
-absl::optional<Http::Code>
-FaultFilter::abortHttpStatus(const Http::RequestHeaderMap& request_headers) {
+AbortHttpAndGrpcStatus FaultFilter::abortStatus(const Http::RequestHeaderMap& request_headers) {
   if (!isAbortEnabled(request_headers)) {
-    return absl::nullopt;
+    return std::make_pair(absl::nullopt, absl::nullopt);
   }
 
+  auto grpc_status = abortGrpcStatus(request_headers);
+
+  // If gRPC status code is set, then HTTP will be set to 200
+  return grpc_status.has_value() ? std::make_pair(Http::Code::OK, grpc_status)
+                                 : std::make_pair(abortHttpStatus(request_headers), grpc_status);
```

Comment:
Can we pass `absl::nullptr` instead of `grpc_status` here to make it clearer that we don't return GRPC status code when we return an HTTP status code from `abortHttpStatus(request_headers)` ?